src/sb_pointcloud_processing/src/test_pcl_generator_node.cpp

```diff
@@ -0,0 +1,125 @@
+/*
+ * Created by: Min Gyo Kim
+ * Created On: March 11th 2018
+ * Description: Generates a point cloud with two lines and an optional outlier
+ * line
+ *              and publishes it to "/input/pointcloud"
+ */
+
+#include "../test/TestUtils.h"
+#include <pcl/point_cloud.h>
+#include <pcl_conversions/pcl_conversions.h>
+#include <ros/ros.h>
+#include <sb_utils.h>
+#include <sensor_msgs/PointCloud2.h>
+
+std::vector<float> first_line;
+std::vector<float> second_line;
+
+float x_min;
+float x_max;
+float x_delta;
+float max_noise_x;
+float max_noise_y;
+
+float outlier_x_delta;
+std::vector<float> outlier_line;
+
+sensor_msgs::PointCloud2 generatePclMessage(bool include_outlier);
+
+int main(int argc, char** argv) {
+    ros::init(argc, argv, "test_pcl_generator_node");
+    ros::NodeHandle nh;
+    ros::NodeHandle private_nh("~");
+
+    ros::Rate loop_rate = 0.75;
+    ros::Publisher publisher =
+    nh.advertise<sensor_msgs::PointCloud2>("input_pointcloud", 1);
+
+    std::string first_line_param          = "first_line";
+    std::vector<float> default_first_line = {50, 0, -0.01};
+    private_nh.param(first_line_param, first_line, default_first_line);
+
+    std::string second_line_param          = "second_line";
+    std::vector<float> default_second_line = {0, 0, -0.01};
+    private_nh.param(second_line_param, second_line, default_second_line);
+
+    std::string x_min_param = "x_min";
+    float default_x_min     = -50;
+    SB_getParam(private_nh, x_min_param, x_min, default_x_min);
+
+    std::string x_max_param = "x_max";
+    float default_x_max     = 50;
```

Comment:
Bit of a nit-pick, but I might suggest choosing more realistic default values in terms of size (perhaps 5m instead of 50m)?

Reply:
done